presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorMTlsTest.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <arrow/flight/types.h>
+#include <arrow/testing/gtest_util.h>
+#include <folly/init/Init.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include "presto_cpp/main/connectors/arrow_flight/ArrowFlightConnector.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/ArrowFlightConnectorTestBase.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/ArrowFlightPlanBuilder.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/Utils.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/config/Config.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+
+using namespace arrow;
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+
+namespace facebook::presto::test {
+
+class ArrowFlightConnectorMtlsTestBase : public ArrowFlightConnectorTestBase {
+ protected:
+  explicit ArrowFlightConnectorMtlsTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : ArrowFlightConnectorTestBase(std::move(config)) {}
+
+  void setFlightServerOptions(
+      flight::FlightServerOptions* serverOptions) override {
+    flight::CertKeyPair tlsCertificate{
+        .pem_cert = readFile("./data/mtls_certs/server.crt"),
+        .pem_key = readFile("./data/mtls_certs/server.key")};
+    serverOptions->tls_certificates.push_back(tlsCertificate);
+    serverOptions->verify_client = true;
+    serverOptions->root_certificates = readFile("./data/mtls_certs/ca.crt");
+  }
+
+  void executeTest(
+      bool isPositiveTest = true,
```

Comment:
you don't need this flag, just use `VELOX_ASSERT_THROW` directly in the test itself